pkg/api/api.proto

```diff
@@ -2,37 +2,148 @@ syntax = "proto3";
+        };
+    };
+    rpc GetSuggestions(GetSuggestionsRequest) returns (GetSuggestionsReply){
+        option (google.api.http) = {
+            post: "/api/Manager/GetSuggestions"
+            body: "*"
+        };
+    };
+    rpc GetShouldStopWorkers(GetShouldStopWorkersRequest) returns (GetShouldStopWorkersReply){
+        option (google.api.http) = {
+            post: "/api/Manager/GetTrials/GetShouldStopWorkers"
+            body: "*"
+        };
+    };
+    rpc GetMetrics(GetMetricsRequest) returns (GetMetricsReply){
+        option (google.api.http) = {
+            post: "/api/Manager/GetMetrics"
+            body: "*"
+        };
+    };
+    rpc SetSuggestionParameters(SetSuggestionParametersRequest) returns (SetSuggestionParametersReply){
+        option (google.api.http) = {
+            post: "/api/Manager/SetSuggestionParameters"
+            body: "*"
+        };
+    };
+    rpc GetSuggestionParameters(GetSuggestionParametersRequest) returns (GetSuggestionParametersReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetSuggestionParameters/{param_id}"
+        };
+    };
+    rpc GetSuggestionParameterList(GetSuggestionParameterListRequest) returns (GetSuggestionParameterListReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetSuggestionParameterList/{study_id}"
+        };
+    };
+    rpc SetEarlyStoppingParameters(SetEarlyStoppingParametersRequest) returns (SetEarlyStoppingParametersReply){
+        option (google.api.http) = {
+            post: "/api/Manager/SetEarlyStoppingParameters"
+            body: "*"
+        };
+    };
+    rpc GetEarlyStoppingParameters(GetEarlyStoppingParametersRequest) returns (GetEarlyStoppingParametersReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetEarlyStoppingParameters/{param_id}"
+        };
+    };
+    rpc GetEarlyStoppingParameterList(GetEarlyStoppingParameterListRequest) returns (GetEarlyStoppingParameterListReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetEarlyStoppingParameterList/{study_id}"
+        };
+    };
+    rpc SaveStudy(SaveStudyRequest) returns(SaveStudyReply){
+        option (google.api.http) = {
+            post: "/api/Manager/SaveStudyReply"
+            body: "*"
+        };
+    };
+    rpc SaveModel(SaveModelRequest) returns(SaveModelReply){
+        option (google.api.http) = {
+            post: "/api/Manager/SaveModel"
+            body: "*"
+        };
+    };
+    rpc GetSavedStudies(GetSavedStudiesRequest) returns(GetSavedStudiesReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetSavedStudies"
+        };
+    };
+    rpc GetSavedModels(GetSavedModelsRequest) returns(GetSavedModelsReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetSavedModels"
+        };
+    };
 }
 
 service Suggestion {
-    rpc GetSuggestions(GetSuggestionsRequest) returns (GetSuggestionsReply);
```

Comment:
Do we need rest for suggestion and earlystopping service?

Reply:
thanks for pointing it out! it is not required.